src/Build/BackEnd/Components/Logging/LoggingService.cs

```diff
@@ -1175,10 +1171,10 @@ public void LogBuildEvent(BuildEventArgs buildEvent)
                 {
                     // Only log certain events if OnlyLogCriticalEvents is true
                     if (
-                        (warningEvent != null)
-                        || (errorEvent != null)
-                        || (buildEvent is CustomBuildEventArgs)
-                        || (buildEvent is CriticalBuildMessageEventArgs)
+                        buildEvent is BuildWarningEventArgs
```

Comment:
Notice that we're now doing the type check twice in this method where we only did one previously. We check if something is a warning, then check it again. I'm sure it's fine and we won't notice the perf difference, but it's important to be aware of these things in general. Again, I'm not worried about this place in particular.

Reply:
Good thing to point out. I'd originally written it with just one type check but cleaner, and my code didn't compile. I ended up going with cleaner over more performant here.